src/main/java/edu/hm/hafner/analysis/parser/pvsstudio/AnalyzerType.java

```diff
@@ -0,0 +1,148 @@
+     */
+    public static final int CUSTOMERSPECIFIC_CCPP_ERRORCODE_END = 2499;
+    /**
+     * MISRA errors.
+     * https://www.viva64.com/en/w/#MISRA
+     */
+    public static final int MISRA_CCPP_ERRORCODE_BEGIN = 2500;
+    /**
+     * MISRA errors.
+     * https://www.viva64.com/en/w/#MISRA
+     */
+    public static final int MISRA_CCPP_ERRORCODE_END = 2999;
+    /**
+     * General Analysis (C#).
+     * https://www.viva64.com/en/w/#GeneralAnalysisCS
+     */
+    public static final int GENERAL_CS_ERRORCODE_BEGIN = 3000;
+    /**
+     * General Analysis (C#).
+     * https://www.viva64.com/en/w/#GeneralAnalysisCS
+     */
+    public static final int GENERAL_CS_ERRORCODE_END = 3499;
+
+    /**
+     * General Analysis (Java).
+     * https://www.viva64.com/en/w/#GeneralAnalysisJAVA
+     */
+    public static final int GENERAL_JAVA_ERRORCODE_BEGIN = 6000;
+    /**
+     * General Analysis (Java).
+     * https://www.viva64.com/en/w/#GeneralAnalysisJAVA
+     */
+    public static final int GENERAL_JAVA_ERRORCODE_END = 6999;
+
+    /**.
+     * @param errorCodeStr - error code
+     * @return anayzer type
+     */
+    public static AnalyzerType getAnalyzerType(final String errorCodeStr) {
+        if (errorCodeStr == null ||  errorCodeStr.length() <= 1) {
+            return AnalyzerType.UNKNOWN;
+        }
+
+        try {
+            String sub = errorCodeStr.substring(1);
+
+            int errorCode = Integer.parseInt(sub);
+
+            if (errorCode >= VIVA64_CCPP_ERRORCODE_BEGIN && errorCode <= VIVA64_CCPP_ERRORCODE_END) {
+                return AnalyzerType.VIVA_64;
+            }
+            else if (errorCode >= GENERAL_CCPP_LOW_ERRORCODE_BEGIN && errorCode <= GENERAL_CCPP_LOW_ERRORCODE_END) {
+                return AnalyzerType.GENERAL;
+            }
+            else if (errorCode >= OPTIMIZATION_CCPP_ERRORCODE_BEGIN && errorCode <= OPTIMIZATION_CCPP_ERRORCODE_END) {
+                return AnalyzerType.OPTIMIZATION;
+            }
+            else if (errorCode >= GENERAL_CCPP_HIGH_ERRORCODE_BEGIN && errorCode <= GENERAL_CCPP_HIGH_ERRORCODE_END) {
+                return AnalyzerType.GENERAL;
+            }
+            else if (errorCode >= CUSTOMERSPECIFIC_CCPP_ERRORCODE_BEGIN && errorCode <= CUSTOMERSPECIFIC_CCPP_ERRORCODE_END) {
+                return AnalyzerType.CUSTOMER_SPECIFIC;
+            }
+            else if (errorCode >= MISRA_CCPP_ERRORCODE_BEGIN && errorCode <= MISRA_CCPP_ERRORCODE_END) {
+                return AnalyzerType.MISRA;
+            }
+            else if (errorCode >= GENERAL_CS_ERRORCODE_BEGIN && errorCode <= GENERAL_CS_ERRORCODE_END) {
+                return AnalyzerType.GENERAL;
+            }
+            else if (errorCode >= GENERAL_JAVA_ERRORCODE_BEGIN && errorCode <= GENERAL_JAVA_ERRORCODE_END) {
+                return AnalyzerType.GENERAL;
+            }
+        }
+        catch (StringIndexOutOfBoundsException | NumberFormatException ex) {
```

Comment:
`StringIndexOutOfBoundsException` is never thrown. Anyway with `IntegerParser` you get everything or free